### ISSUE TITLE
[HL] Camera ShaderFilters removes all black from cameras on top of it

### ISSUE THREAD

**Issue by ThePlank** [HaxeFlixel/flixel#2513]
- **Haxe version:** 4.1.5
- **Flixel version:** 4.10.0
- **OpenFL version:** 9.1.0
- **Lime version:** 7.9.0
- **Affected targets:** Hashlink

________________________________

**Code snippet reproducing the issue**:

Using the [Filters](https://github.com/HaxeFlixel/flixel-demos/tree/dev/Effects/Filters/source) demo

________________________________

**Observed behavior:**
The Filter appears to remove all black from the cameras on top
![image](https://user-images.githubusercontent.com/75984453/156892011-00c6f74a-23cf-45a2-ab9e-94350f60f4b8.png)

**Expected behavior:**
FIlter works as expected like in other targets (eg. HTML5)
Using "Apply to full game" as a example:
![image](https://user-images.githubusercontent.com/75984453/156892074-937d67ba-385b-43e1-96e0-115c9128d45a.png)

**Comment by ThePlank** [comment]
nevermind its because of openfl's ShaderFilter blendmode gets set to ADD on hashlink (i know im necroposting but ovjfhovefhih)

**Comment by Geokureli** [comment]
you have my interest and attention. this still seems like something that should be fixed

**Comment by ACrazyTown** [comment]
@Geokureli 

I created a little test project in which you can reproduce the issue (https://github.com/ACrazyTown/flixel-hl-shader-bug)

When a camera that has a shader filter is attached to a `FlxSprite` via the `cameras` variable (maybe the `camera` variable too, haven't tested), substates that contain black will turn translucent. It also affects the debug menu.

Example with a sprite that doesn't have a camera attached, everything's fine.
![image](https://user-images.githubusercontent.com/47027981/231298446-52c09951-1644-449a-97db-f1e2fc28f707.png)

 Example with a sprite that has the camera attached, the black spot is gone and the debug menu appears brighter
![image](https://user-images.githubusercontent.com/47027981/231299113-1d3880a0-2ce9-4881-ab1e-26ab2ce0855e.png)

**Comment by ThePlank** [comment]
nvm i was wrong
the blend mode fix is only for the camera its being applied in (because that was also happening on the camera itself)
adding this code
```hx
for (filter in filters) {
	if (filter is ShaderFilter)
		cast (filter, ShaderFilter).blendMode = NORMAL;
}
```
still makes it do this terribleness
![image](https://user-images.githubusercontent.com/75984453/232130880-4ffe7a91-62e7-4cf8-868a-f36c537078b6.png)

but i found out that it doesent do that when the window's unfocused (may take a bit to load because gif)
![Filters_MzsOUJ9Wu1](https://user-images.githubusercontent.com/75984453/232131362-f3cfa1e3-11af-4f71-a69d-ad031dfb79d2.gif)

**Comment by ThePlank** [comment]
updated the title to fix my terrible english (i am terrible at english)

**Comment by ThePlank** [comment]
holy shit how did i not notice that
<img src="https://user-images.githubusercontent.com/75984453/232232891-fcc8dbe9-d18f-4267-8158-474097565764.jpg" width="25%"/>

**Comment by ThePlank** [comment]
fixed by openfl/openfl#2619
